packages/frontend/src/hooks/useEns.ts

```diff
@@ -1,33 +1,49 @@
+import { isAddress } from 'ethers/lib/utils'
 import { useEffect, useState } from 'react'
 import logger from 'src/logger'
-import Address from 'src/models/Address'
-import { getEnsAvatar, getEnsName } from 'src/utils/ens'
+import { getEnsAddress, getEnsAvatar, getEnsName } from 'src/utils/ens'
 
-export function useEns(address?: Address) {
+export function useEns(addressOrEnsName?: string) {
   const [ensAvatar, setEnsAvatar] = useState<string>()
-  const [ensName, setEnsName] = useState<string>()
+  const [ensName, setEnsName] = useState<string | null>()
+  const [ensAddress, setEnsAddress] = useState<string>()
 
   useEffect(() => {
-    if (address?.address) {
-      try {
-        getEnsName(address.address).then(en => en && setEnsName(en))
-      } catch (err) {
-        logger.error(`error during setEnsName:`, err)
+    setEnsName('')
+    setEnsAddress('')
+    setEnsAvatar('')
+
+    if (!addressOrEnsName) {
+      return
+    }
+
+    try {
+      if (isAddress(addressOrEnsName)) {
+        getEnsName(addressOrEnsName).then(setEnsName)
```

Comment:
shouldn't there be an `await` here in order for it to be caught in try/catch? otherwise it throws an `Uncaught in promise` error

error uncaught without await

![DeepinScreenshot_select-area_20220113175714](https://user-images.githubusercontent.com/168240/149438322-414d662c-4d25-4bbc-8b60-55f6db69e44b.png)

error caught with await

![DeepinScreenshot_select-area_20220113175751](https://user-images.githubusercontent.com/168240/149438328-f4b79f5f-125a-4f11-8677-a72dd87b2e4c.png)